### PR TITLE
RavenDB-7787a Add Certificate (Security) and DatabaseRecord configuration

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Embedded/RavenEmbedded.cs [hunk 1]
```diff
@@ -35,6 +39,29 @@ public void StartServer(ServerOptions options = null)
...
+            if (options.Security != null)
+            {
+                _certificate = options.Security.ClientCertificate;
+
+                try
+                {
+                    var thumbprint = options.Security.ServerCertificiateThumbprint;
+                    RequestExecutor.ServerCertificateCustomValidationCallback +=
+                        (message, certificate2, chain, errors) =>
+                            certificate2.Thumbprint == thumbprint;
```
Comment (maximburyak) [inline src/Raven.Embedded/RavenEmbedded.cs:51]: Shoud be configurable. What if we have several thumbprints etc? ServerCertificateCustomValidationCallback is static

File: src/Raven.Embedded/DatabaseOptions.cs [hunk 2]
```diff
@@ -1,9 +1,13 @@
-﻿namespace Raven.Embedded
+﻿using Raven.Client.ServerWide;
+
+namespace Raven.Embedded
 {
     public class DatabaseOptions
     {
         public string DatabaseName { get; set; }
```
Comment (ppekrol) [inline src/Raven.Embedded/DatabaseOptions.cs:7]: Now when we have DatabaseRecord then I'm not sure if we need this. We should drop this or change DatabaseRecord to `Action<DatabaseRecord>` to modify the record which database name was set using the `DatabaseName` property.

File: src/Raven.Embedded/RavenEmbedded.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Embedded/RavenEmbedded.cs:51]: Not sure if this is something that we need to support with embedded? Only one instance of server will be started by application.

Comment (IdanHaim) [reply src/Raven.Embedded/RavenEmbedded.cs:51]: The server still needs to be trusted

Comment (ayende) [reply src/Raven.Embedded/RavenEmbedded.cs:51]: And this add trust to the server.
User can also configure this separately from us, which will overwrite this logic.

File: src/Raven.Embedded/DatabaseOptions.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Embedded/DatabaseOptions.cs:7]: I like the action better, yes.

**Comment by ayende** [comment]
@ppekrol ` certificate2.Thumbprint == thumbprint;` is required when running HTTPS, otherwise, how would we know that the server is trustworthy?
By definition, almost, we aren't going to be using a valid cert, since we are bound to `127.0.0.1`.

**Comment by ppekrol** [comment]
I was answering to @maximburyak

**Pull request merged**